site/content/docs/11-deploying.md

```diff
@@ -4,28 +4,25 @@ title: Deployment
 
 Sapper apps run anywhere that supports Node 8 or higher.
 
+### Deploying to Vercel ([formerly ZEIT Now](https://vercel.com/blog/zeit-is-now-vercel))
 
-### Deploying to Now
+#### Using Now v2
 
-> This section relates to Now 1, not Now 2
+We can use an unoffical third-party libary like [now-sapper loader](https://www.npmjs.com/package/now-sapper) to deploy our projects to [Vercel] and even use our sapper runtime environment there.
```

Comment:
Why do we need to use an unofficial third-party library instead of doing `npm i -g vercel`?

Reply:
Now (and also, I would assume, Vercel) does not support dynamic Sapper applications. They only natively support exported Sapper apps, the sort that you could host on any static file host. https://github.com/zeit/now-examples/issues/193#issuecomment-517802287 Thus the third-party library. I have not tried it, but my understanding is that it does not separate the different routes into their own files, which is something that a proper Now (Vercel?) builder ought to do.

Reply:
@benmccann you still need the `vercel` cli regardless. `now-sapper` is a builder. As @Conduitry says `vercel/zeit/now/whatever it's called` have no explicit for Sapper apps, simply the static apps that Sapper can build.

`now-sapper` gives us that support, albeit with one (minor) tradeoff.

Reply:
I don't quite understand why Vercel doesn't support Sapper by default. Is it because Sapper doesn't run on Lambda (https://github.com/sveltejs/sapper/issues/356)? If so, I sent a PR that I think would enable that: https://github.com/sveltejs/sapper/pull/1215.

Reply:
> you still need the vercel cli regardless

We should document that. If I go to https://github.com/thgh/now-sapper it doesn't tell you how to deploy the project after it's built. It says how to configure the project, but not what commands to run to build or deploy

Anyway, I'm still not really understanding why a third-party tool is required. Vercel has a [Svelte sample](https://github.com/zeit/now/tree/master/examples/svelte) that includes server-side code. When I visit their deployed sample I can see that it does a `fetch` call in the Chrome debugger.

Maybe we should just link to [their instructions](https://vercel.com/guides/deploying-svelte-with-vercel) instead?

Reply:
I hadn't seen that example before, but I do also notice that it is a _Svelte_ app, not a Sapper one. The server-side code is just individual files that CJS-export a `(req, res)` object to handle requests to that URL - it's not related at all to Svelte, which seems to be running entirely on the client side.

Doing a proper deployment to Now/Vercel of a dynamic SSR'd Sapper app would presumably involve generating a separate lambda for each page of the app, similar to the `api/date.js` in the Svelte example.

Reply:
@benmccann their examples don't work for Sapper, that's why we built `now-sapper`. I don't know the technical details of it, but feel free to try to deploy a Sapper app without the builder if you understand how to diagnose the issues and document a simpler approach. You'll notice that `next` itself has a `now-next` builder, so writing a builder does seem to be the official approach.

With regards to deployment instructions, the view was that you'd find Now and then want a builder, so you'd know that the deployment for Vercel is as simple as typing three letters: `now`. Or I guess `vercel` today.

I agree, this is something we should document though.

Reply:
Thanks for bearing with me on this and providing the additional info. This looks good to me. I sent a PR to update the README in the other project: https://github.com/thgh/now-sapper/pull/28